lang/c++/impl/DataFile.cc

```diff
@@ -522,18 +522,18 @@ void DataFileReaderBase::sync(int64_t position) {
     DataFileSync sync_buffer;
     const uint8_t *p = nullptr;
     size_t n = 0;
-    size_t i = 0;
+    int i = 0;
```

Comment:
It makes sense to keep `i` a `size_t` because it is used for indexing and compared against the other `size_t` `SyncSize`.